pkg/ui/app/containers/timescale.tsx

```diff
@@ -86,50 +144,42 @@ class TimeScaleSelector extends React.Component<TimeScaleSelectorProps, TimeScal
   }
 
   componentWillMount() {
-    // Hide the popup when you click anywhere on the page
-    document.body.addEventListener("click", this.hide);
     this.props.refreshNodes();
     this.setDefaultTime();
   }
 
-  componentWillReceiveProps(props: TimeScaleSelectorProps) {
+  componentWillReceiveProps(props: TimeScaleDropdownProps) {
     if (!props.nodeStatusesValid) {
       this.props.refreshNodes();
     } else {
       this.setDefaultTime(props);
     }
   }
 
-  componentWillUnmount() {
-    // Remove popup hiding event listener on unmount
-    document.body.removeEventListener("click", this.hide);
-  }
-
   render() {
-    let selectorClass = classNames({
-      "timescale-selector": true,
-      "show": this.state.controlsVisible,
-    });
-    return <div className="timescale-selector-container">
-      <button
-        className="timescale"
-        ref={(timescaleBtn) => this.timescaleBtn = timescaleBtn}
-        onClick={() => this.setVisible(!this.state.controlsVisible)}>
-          Select Timescale
-      </button>
-      <div className={selectorClass}>
-        <div className="text">View Last: </div>
-        {
-          _.map(this.props.availableScales, (scale, key) => {
-            let theseSettings = scale;
-            return <button key={key}
-                           className={classNames({selected: this.isSelected(scale)})}
-                           onClick={(e) => { this.changeSettings(theseSettings); this.setVisible(false); } }>
-                           { key }
-            </button>;
-          })
-        }
-      </div>
+    let currentScale = this.props.currentScale;
+    let timeRange: React.ReactNode = "";
+
+    let className = "Select--side-arrows";
```

Comment:
This is a really odd decision. "Select" is the name of the CSS class applied to the react-select component.  You've created html here that looks like:

```html
<div class="Select--side-arrows">
  <div class="dropdown">
     <div class="Select dropdown__select">
```

That's very difficult to comprehend.  

Why isn't side arrow visibility handled by the dropdown component?  You even added an "arrows" boolean property to the dropdown, but then didn't use it, instead electing to put the arrows on every single dropdown but keep them hidden, and then overriding that hidden behavior using this oddly-named CSS class in the time window component.